tensorboard/webapp/hparams/_redux/hparams_effects_test.ts

```diff
@@ -0,0 +1,206 @@
+import * as appRoutingActions from '../../app_routing/actions';
+import * as coreActions from '../../core/actions';
+import {RouteKind} from '../../app_routing/types';
+
+function mockHparamsDataSourceFactory() {
+  return jasmine.createSpyObj<HparamsDataSource>('HparamsDataSource', [
+    'fetchExperimentInfo',
+    'fetchSessionGroups',
+  ]);
+}
+
+describe('hparams effects', () => {
+  let dataSource: jasmine.SpyObj<HparamsDataSource>;
+  let store: MockStore<State>;
+  let effects: HparamsEffects;
+  let action: ReplaySubject<Action>;
+  let dispatchSpy: jasmine.Spy;
+  let actualActions: Action[];
+
+  let mockHparamsAndMetricsSpecs: HparamAndMetricSpec;
+  let mockSessionGroups: SessionGroup[];
+
+  beforeEach(async () => {
+    action = new ReplaySubject<Action>(1);
+
+    await TestBed.configureTestingModule({
+      providers: [
+        provideMockActions(action),
+        HparamsEffects,
+        provideMockTbStore(),
+        {provide: HparamsDataSource, useFactory: mockHparamsDataSourceFactory},
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+
+    actualActions = [];
+    dispatchSpy = spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      actualActions.push(action);
+    });
+
+    effects = TestBed.inject(HparamsEffects);
+    dataSource = TestBed.inject(
+      HparamsDataSource
+    ) as jasmine.SpyObj<HparamsDataSource>;
+
+    mockHparamsAndMetricsSpecs = {
+      hparams: [buildHparamSpec({name: 'h1'})],
+      metrics: [buildMetricSpec({tag: 'm1'})],
+    };
+    mockSessionGroups = [
+      {
+        name: 'session_group_1',
+        hparams: {
+          hparam1: 1,
+          hparam2: 'abc',
+        },
+        sessions: [
+          {
+            name: 'exp1/run1',
+            status: RunStatus.STATUS_UNKNOWN,
+            metricValues: [] as any,
+            modelUri: '',
+            monitorUrl: '',
+            startTimeSecs: 123,
+            endTimeSecs: 456,
+          },
+        ],
+      },
+    ];
+
+    dataSource.fetchExperimentInfo.and.returnValue(
+      of(mockHparamsAndMetricsSpecs)
+    );
+
+    dataSource.fetchSessionGroups.and.returnValue(of(mockSessionGroups));
+  });
+
+  describe('loadHparamsData$', () => {
+    beforeEach(() => {
+      effects.loadHparamsData$.subscribe((action) => {
+        actualActions.push(action);
+      });
+      store.overrideSelector(selectors.getEnableHparamsInTimeSeries, true);
+      store.overrideSelector(selectors.getActiveRoute, {
+        routeKind: RouteKind.EXPERIMENT,
+        params: {},
+      });
+      store.overrideSelector(selectors.getExperimentIdsFromRoute, ['exp1']);
+      store.refreshState();
+    });
+
+    it('does not dispatch requests when enableHparamsInTimeSeries is false', () => {
+      store.overrideSelector(selectors.getEnableHparamsInTimeSeries, false);
+      store.refreshState();
+
+      action.next(runsActions.runTableShown({experimentIds: ['exp1']}));
```

Comment:
Using 'exp1' (the same as getExperimentIdsFromRoute) doesn't make it clear where the code is getting the experiment ids from. Can we choose something different? (perhaps this suggestion is obsolete if you change getExperimentIdsFromRoute to return 'expFromRoute' instead.)